src/components/navbars/ProfileNav.jsx

```diff
@@ -1,14 +1,47 @@
-import { Link } from "react-router-dom";
+import { Link } from 'react-router-dom';
+import { logout } from '../../util/DataBaseRequests';
 import logoutIcon from '../../assets/icons/logout.png';
+import { useAuth } from '../../AuthProvider';
+import { useNavigate } from 'react-router-dom';
+import { useState } from 'react';
 
 const ProfileNav = () => {
-    return(
-            <nav className="sm:hidden relative flex flex-row my-4 gap-2 items-center">
-                <div id="user-photo" className="h-10 w-10 rounded-full bg-lightGreen"></div>
-                <p className="font-spartan text-lg font-semibold">User name</p>
-                <Link to="/"><img src={logoutIcon} className="h-6" alt="logout icon"/></Link>
-            </nav>
-    )
+  const [logoutError, setLogoutError] = useState({});
+  const { clearUserSession, userData } = useAuth();
+
+  const navigate = useNavigate();
+  const handleLogout = async () => {
+    try {
+      const result = await logout();
+      if (result.status === 200) {
+        clearUserSession();
+        navigate('/');
+      }
+    } catch (error) {
+      console.log(error);
+      setLogoutError({
```

Comment:
Is it necessary to spread this new error object? Is it not just re-writing the empty object?